std-bits/base/src/main/java/org/enso/base/enso_cloud/logging/LogApiAccess.java

```diff
@@ -0,0 +1,230 @@
+package org.enso.base.enso_cloud.logging;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import org.enso.base.enso_cloud.AuthenticationProvider;
+
+/**
+ * Gives access to the low-level log event API in the Cloud and manages asynchronously submitting
+ * the logs.
+ */
+public abstract class LogApiAccess {
+  /**
+   * We still want to limit the batch size to some reasonable number - sending too many logs in one
+   * request could also be problematic.
+   */
+  private static final int MAX_BATCH_SIZE = 100;
+
+  private static final int MAX_RETRIES = 5;
+  private static final Logger LOGGER = Logger.getLogger(LogApiAccess.class.getName());
+
+  private HttpClient httpClient;
+  private final LogJobsQueue logQueue = new LogJobsQueue();
+  private final ThreadPoolExecutor backgroundThreadService;
+  private RequestConfig cachedRequestConfig = null;
+
+  protected LogApiAccess() {
+    // We set-up a thread 'pool' that will contain at most one thread.
+    // If the thread is idle for 60 seconds, it will be shut down.
+    backgroundThreadService =
+        new ThreadPoolExecutor(0, 1, 60L, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
+  }
+
+  public abstract URI endpoint();
```

Comment:
This method should be `protected abstract` - e.g. to clarify that it is supposed to be [implemented by subclasses](http://wiki.apidesign.org/wiki/ClarityOfAccessModifiers#Meaning_of_protected_abstract).